### PR TITLE
[GEOS-7822] Support unset primitive fields on REST PUT against ServiceInfo

### PR DESCRIPTION
**Pull request by tbarsballe**
https://osgeo-org.atlassian.net/browse/GEOS-7822

This fix uses the same approach as in https://github.com/geoserver/geoserver/pull/1541, but for ServiceInfo instead of CatalogInfo this time around

### CODE REVIEW

**Comment by tbarsballe** [comment]
Part of this fix adds the getObjectClass() to method, in order to obtain what type of ServiceInfo is being converted. This is necessary because gs-wms is downstream of gs-main, meaning gs-main is unaware of the WMSInfo class.

For consistency, I have also modified AbstractCatalogInfoConverter with the same implementation (as a seperate commit).

File: src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java [hunk 1]
```diff
@@ -2291,4 +2278,39 @@ public Object doUnmarshal(Object result,
...
+        public Object doUnmarshal(Object result,
+                HierarchicalStreamReader reader, UnmarshallingContext context) {
+            ServiceInfoImpl obj = (ServiceInfoImpl) super.doUnmarshal(result, reader, context);
+            
+            return obj;
+        }
+        @Override
+        protected Object instantiateNewInstance(HierarchicalStreamReader reader,
+                UnmarshallingContext context) {
+            Object emptyObject = super.instantiateNewInstance(reader, context);
```
Comment (jodygarnett) [inline src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java]: This is where the bug is actually fixed, can we get a javadoc or something to show how important this logic is

File: src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java [hunk 2]
```diff
@@ -2291,4 +2278,39 @@ public Object doUnmarshal(Object result,
             return obj;
         }
     }
+    
+    /**
+     * Converter for ServiceInfo class
+     */
+    public class ServiceInfoConverter extends AbstractReflectionConverter {
```
Comment (jodygarnett) [inline src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java]: Can we have some more details for this new public class?

**Comment by tbarsballe** [comment]
Javadocs updated

**Comment by tbarsballe** [comment]
Backported to 2.10.x.

**Pull request merged**